src/Analysis/GWASResultsV2/Views/Home/HomeTable/HomeTable.jsx

```diff
@@ -1,94 +1,295 @@
-import React, { useContext } from 'react';
-import { Button, Table, Space } from 'antd';
+import React, { useContext, useEffect, useState } from 'react';
+import {
+  Button, Table, Space, Input, DatePicker, Select,
+} from 'antd';
+import { SearchOutlined } from '@ant-design/icons';
 import PropTypes from 'prop-types';
+import moment from 'moment';
 import SharedContext from '../../../Utils/SharedContext';
 import ActionsDropdown from './ActionsDropdown/ActionsDropdown';
 import Icons from './TableIcons/Icons';
 import DateForTable from '../../../SharedComponents/DateForTable/DateForTable';
 import PHASES from '../../../Utils/PhasesEnumeration';
+import filterTableData from './filterTableData';
+import VIEWS from '../../../Utils/ViewsEnumeration';
 import './HomeTable.css';
 
+const { RangePicker } = DatePicker;
+
 const HomeTable = ({ data }) => {
-  const { setCurrentView, setSelectedRowData } = useContext(SharedContext);
+  const {
+    setCurrentView,
+    setSelectedRowData,
+    homeTableState,
+    setHomeTableState,
+  } = useContext(SharedContext);
+
+  const handleTableChange = (pagination, filters, sorter) => {
+    if (pagination.current !== homeTableState.currentPage) {
+      // User changes page selection, set page to current pagination selection
+      return setHomeTableState({
+        ...homeTableState,
+        currentPage: pagination.current,
+      });
+    }
+    // When the user updates sorting or filtering, set page to first page
+    return setHomeTableState({
+      ...homeTableState,
+      currentPage: 1,
+      sortInfo: sorter,
+    });
```

Comment:
question: if this already takes care of setting to page 1, why is it repeated in the handlers below?